feat: complete events.Event docs (#35360)
mdn/content: files/en-us/mozilla/add-ons/webextensions/api/events/event/getrules/index.md

---
title: Event.getRules()
slug: Mozilla/Add-ons/WebExtensions/API/events/Event/getRules
page-type: webextension-api-event
browser-compat: webextensions.api.events.Event.getRules
---

{{AddonSidebar}}

Retrieves all the rules registered for the event.

## Syntax

```js-nolint
events.Event.getRules(ruleIdentifiers, callback)
```

### Parameters

- `ruleIdentifiers` {{optional_inline}}

>   - : `array` of `string`. If an array of `string` is passed, only rules with identifiers contained in this array are returned, otherwise all rules registered are returned.

Reviewer: ```suggestion
  - : `array` of `string`. Identifiers for the rules to return. If no array is passed, all rules are returned.
```

- `callback`

>   - : `function`. The parameter of the callback function is an array of `events.Rule` objects that are currently registered.

Reviewer: ```suggestion
  - : `function`. The parameter of the callback function is an array of {{WebExtAPIRef("events.Rule")}} objects.
```
